specification/compute/resource-manager/Microsoft.Compute/stable/2021-07-01/examples/CreateOrUpdateAVmWithCapacityReservation.json

```diff
@@ -0,0 +1,189 @@
+              }
+            }
+          ]
+        }
+      }
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "name": "myVM",
+        "properties": {
+          "osProfile": {
+            "adminUsername": "{your-username}",
+            "secrets": [],
+            "computerName": "myVM",
+            "windowsConfiguration": {
+              "provisionVMAgent": true,
+              "enableAutomaticUpdates": true
+            }
+          },
+          "capacityReservation": {
+            "capacityReservationGroup": {
+              "id": "/subscriptions/{subscription-id}/resourceGroups/myResourceGroup/providers/Microsoft.Compute/CapacityReservationGroups/{crgName}"
+            }
+          },
+          "networkProfile": {
+            "networkInterfaces": [
+              {
+                "id": "/subscriptions/{subscription-id}/resourceGroups/myResourceGroup/providers/Microsoft.Network/networkInterfaces/nsgExistingNic",
+                "properties": {
+                  "primary": true
+                }
+              }
+            ]
+          },
+          "storageProfile": {
+            "imageReference": {
+              "sku": "standard-data-science-vm",
+              "publisher": "microsoft-ads",
+              "version": "latest",
+              "offer": "standard-data-science-vm"
+            },
+            "osDisk": {
+              "osType": "Windows",
+              "caching": "ReadOnly",
+              "diffDiskSettings": {
+                "option": "Local"
+              },
+              "managedDisk": {
+                "storageAccountType": "Standard_LRS"
+              },
+              "createOption": "FromImage",
+              "name": "myVMosdisk"
+            },
+            "dataDisks": []
+          },
+          "vmId": "5c0d55a7-c407-4ed6-bf7d-ddb810267c85",
+          "hardwareProfile": {
+            "vmSize": "Standard_DS1_v2"
+          },
+          "provisioningState": "Creating"
+        },
+        "plan": {
+          "publisher": "microsoft-ads",
+          "product": "standard-data-science-vm",
+          "name": "standard-data-science-vm"
+        },
+        "type": "Microsoft.Compute/virtualMachines",
+        "id": "/subscriptions/{subscription-id}/resourceGroups/myResourceGroup/providers/Microsoft.Compute/virtualMachines/myVM",
+        "location": "westus"
+      }
+    },
+    "201": {
+      "body": {
+        "name": "myVM",
+        "properties": {
+          "osProfile": {
+            "adminUsername": "{your-username}",
+            "secrets": [],
+            "computerName": "myVM",
+            "windowsConfiguration": {
+              "provisionVMAgent": true,
+              "enableAutomaticUpdates": true
+            }
+          },
+          "capacityReservation": {
+            "capacityReservationGroup": {
+              "id": "/subscriptions/{subscription-id}/resourceGroups/myResourceGroup/providers/Microsoft.Compute/CapacityReservationGroups/{crgName}"
```

Comment:
> "id": "/subscriptions/{subscription-id}/resourceGroups/myResourceGroup/providers/Microsoft.Compute/CapacityReservationGroups/{crgName}" [](http://example.com/codeflow?start=13&length=136)

Looks like this update will update the VM as well as reservationGroup behind the scene.
Please note this pattern impacts following control plane components
1. Policy: You will not be able to make policy controlling reservation groups for specific VM's not sure if its a scenario.
2. Resource Graph: ARG index will go out of sync on such operations. Please make sure you work with ARG Team to send notification for such cases.

Reply:
Currently we can update the CRG for existing VM only when the VM is deallocated and it wont change the CRG resource but only updates the VM resource by updating the CRG resource Id for VM.
I dont think we are planning to make policies controlling the reservation groups for specific vm. The vm association with CRG is validated in CRP with conditions like validity of CRG and availability of CR under CRG for the vm based on size , zone and other scenarios.
ARm manages sending notifications to ARG and we have tested the scenarios in updating the CRG after the VM is deallcoated.